optaweb-tsp-planner/src/main/java/org/optaweb/tsp/optawebtspplanner/TspPlannerComponent.java

```diff
@@ -145,22 +183,40 @@ public void bestSolutionChanged(BestSolutionChangedEvent<TspSolution> bestSoluti
     }
 
     public void addPlace(Place place) {
-        AirLocation airLocation = fromPlace(place);
+        RoadLocation location = fromPlace(place);
+        Map<RoadLocation, Double> map = new HashMap<>();
+        location.setTravelDistanceMap(map);
+        for (Location other : tsp.getLocationList()) {
+            RoadLocation toLocation = (RoadLocation) other;
+            GHRequest there = new GHRequest(
+                    location.getLatitude(),
+                    location.getLongitude(),
+                    toLocation.getLatitude(),
+                    toLocation.getLongitude());
+            map.put(toLocation, graphHopper.route(there).getBest().getDistance());
+            GHRequest back = new GHRequest(
+                    toLocation.getLatitude(),
+                    toLocation.getLongitude(),
+                    location.getLatitude(),
+                    location.getLongitude());
+            // TODO handle no route -> roll back the problem fact change
```

Comment:
sounds like a good candidate for a test

Reply:
Yes, but since I'm not doing TDD, the code is untestable like this so I need to refactor it first. But I intend to do that as part of this PR.